lib/src/utils.dart

```diff
@@ -89,67 +69,68 @@ List<String> getDependenciesWithPins(Map dependencies) {
   final List<String> infractions = [];
   for (String packageName in dependencies.keys) {
     String version;
+    final packageMeta = dependencies[packageName];
+
     if (dependencies[packageName] is String) {
-      version = dependencies[packageName];
+      version = packageMeta;
     } else {
-      final Map packageMeta = dependencies[packageName];
       if (packageMeta.containsKey('version')) {
         version = packageMeta['version'];
       } else {
         // This feature only works for versions, not git refs or paths.
-        break;
+        continue;
       }
     }
 
-    if (doesVersionPinDependency(version)) {
-      infractions.add(packageName);
+    final DependencyPinEvaluation evaluation = inspectVersionForPins(version);
+
+    if (evaluation.isPin) {
+      infractions.add('$packageName -- ${evaluation.message}');
```

Comment:
Thoughts on also including the bad version range here, for context?

Reply:
I figured they already have the pubspec.yaml